wagtail/locales/views.py

```diff
@@ -42,14 +53,33 @@ class IndexView(generic.IndexView):
             sort_key="language_code",
             url_name="wagtaillocales:edit",
         ),
-        UsageColumn("usage", label=gettext_lazy("Usage")),
+        LocaleUsageColumn("usage", label=gettext_lazy("Usage")),
     ]
 
+    def get_add_url(self) -> Optional[str]:
+        # Removes the "Add" button from the index view.
+        if self.queryset.count() >= len(
+            getattr(settings, "WAGTAIL_CONTENT_LANGUAGES", [])
+        ):
+            return None
+        return super().get_add_url()
+
 
 class CreateView(generic.CreateView):
     page_title = gettext_lazy("Add locale")
     success_message = gettext_lazy("Locale '%(object)s' created.")
 
+    def dispatch(
+        self, request: "HttpRequest", *args: Any, **kwargs: Any
+    ) -> "HttpResponseBase":
+        # Only allow access to the add view if there are locales to be added.
+        if Locale.objects.count() >= len(
+            getattr(settings, "WAGTAIL_CONTENT_LANGUAGES", [])
+        ):
```

Comment:
If there are stale `Locale` records that are not in `WAGTAIL_CONTENT_LANGUAGES`, would this be a problem?

Reply:
good shout 96132be9d8e